packages/core/src/hooks/useRoom.ts

```diff
@@ -77,8 +86,10 @@ export function useRoom(roomOptions?: RoomOptions): RoomState {
           .on(RoomEvent.LocalTrackPublished, onParticipantsChanged)
           .on(RoomEvent.LocalTrackUnpublished, onParticipantsChanged)
           // trigger a state change by re-sorting participants
-          .on(RoomEvent.AudioPlaybackStatusChanged, onParticipantsChanged);
+          .on(RoomEvent.AudioPlaybackStatusChanged, onParticipantsChanged)
+          .on(RoomEvent.StateChanged, onRoomStateChanged);
 
+        await room?.connect(url, token, options);
```

Comment:
@davidzhao I also moved room.connect to the bottom in order to subscribe to all events first.